src/lib.rs

```diff
@@ -58,16 +59,45 @@ use iron::method::Method;
 use iron::status;
 use iron::headers;
 
+/// The struct that builds a CorsMiddleware
+pub struct CorsMiddlewareBuilder {
+    allowed_hosts: Option<HashSet<String>>,
+    allow_credentials: bool
+}
+
+impl CorsMiddlewareBuilder {
+    pub fn new() -> Self {
+        CorsMiddlewareBuilder { allow_credentials: false, allowed_hosts: None }
+    }
+
+    /// Specify which origin hosts are allowed to access the resource.
```

Comment:
Can you add another line here?

    /// If you don't specify any allowed hosts, then any host will be allowed to access the resource.